File: FrontEnd/Affection-App/affection-app/src/app/core/models/requestFilter.ts
import { authResponse } from './authentication/authResponse';

export class RequestFilter {
    gender: string;
    minAge: number = 18;
    maxAge: number = 60;
    pageNumber: number = 1;
    pageSize: number = 5;
    orderBy: string = 'lastActive';
    constructor(user: authResponse) {
        // Set gender filter based on the logged-in user's gender
        this.gender = user.gender === 'Female' ? 'Male' : 'Female';
    }
}

File: BackEnd/Affection-App/Common/Common/RequestFilter.cs

namespace Affection.Contract.Common;
public record RequestFilter
{


    private const int MaxPageSize = 50;

    private int _pageSize = 5;

    public int PageNumber { get; init; } = 1;


    public int PageSize
    {

        get => _pageSize;
        set => _pageSize = (value > MaxPageSize) ? MaxPageSize : value;

    }


    public string UserId { get; set; } = string.Empty;
    public string? Gender { get; set; }
    public string OrderBy { get; set; } = "LastActive";
    public int MinAge { get; set; } = 18;
    public int MaxAge { get; set; } = 60;
}

